sdk/core/System.ServiceModel.Rest/src/Temporary/PipelineProtocolExtensions.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ServiceModel.Rest.Core;
+using System.ServiceModel.Rest.Core.Pipeline;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.ServiceModel.Rest.Shared.Core.Pipeline
+{
+    /// <summary>
+    /// TBD.
+    /// </summary>
+    public static class PipelineProtocolExtensions
```

Comment:
These are extension methods to a type we own, so we don't need to keep them as extension methods in the long run.  I kept them this way for now to make as few changes to the generated code patterns as possible.